Created Grid Component

kingsleyzissou opened the pull request:
Created the grid component.

The grid wrapper is in the 'grid' class.

To create a column add the a child 'column' class and the number of columns ie. 'column six' will create a column that spans six columns. 

Please comment and review.

dimitraz commented on src/_grid.scss:
```diff
@@ -0,0 +1,67 @@
+/* @TODO Move to mixin file
+ * Mixin to create all of the column divisions
+ */
+@mixin column-divisions($prefix, $columns) {
```
what is `prefix` for?

kingsleyzissou replied:
The prefix just adds the prefix before the class name. So for example, for a desktop it adds 'desktop-' to the front of the class name.

That way you can specify specific columns and offsets for different device widths.

dimitraz commented on src/_grid.scss:
```diff
@@ -0,0 +1,67 @@
+
+/* Tablet screens */
+@media screen and (min-width: 700px) and (max-width: 1000px) {
+
```
I would remove these white lines 🐸

kingsleyzissou replied:
Lol cool beans

dimitraz commented on src/_grid.scss:
```diff
@@ -0,0 +1,67 @@
+}
+
+/* Dynamic grid */
+.autofit {
```
I think i preferred `autogrid` more, its more clear. What do you guys think @piccolazucca @kingsleyzissou?

kingsleyzissou replied:
@dimitraz I was thinking of leaving the standard grid in for now, because the autogrid and the standard grid would have different applications.

The autogrid, for example, would be useless to have a sidebar column.

dimitraz commented:
@kingsleyzissou what do you think about have a class for the gutter? like `no-gutter`, `one-gutter` etc or  `gutter-none`, `gutter-two` etc with a default of one or something

kingsleyzissou commented:
@dimitraz yeah that sounds good. I'll have a look to see how we can achieve that

kingsleyzissou commented:
@dimitraz @piccolazucca I've made requested changes to the grid.

I removed the whitespace, added the "gutter-none", "gutter-one" and "gutter-two" helper classes.
I also improved the documentation of the mixins - hopefully those make more sense.

piccolazucca commented on src/_grid.scss:
```diff
@@ -0,0 +1,93 @@
+.column {
+    display: flex;
+    justify-content: center;
+    align-items: center; 
```
Do we want to have this automatically vertically aligned?

piccolazucca commented on src/_grid.scss:
```diff
@@ -0,0 +1,93 @@
+}
+
+/* No gutter helper class */
+.gutter-none {
```
Gutter should be specific to `.grid` only, as it exceeds the `vw` if used with all columns.

kingsleyzissou replied:
@piccolazucca another great spot. Fixed that now too.

piccolazucca commented on src/_grid.scss:
```diff
@@ -0,0 +1,93 @@
+@mixin offset($prefix, $columns) {
+    @each $i in $columns {
+        .#{$prefix}#{'offset-'}#{nth($i, 1)} {
+            grid-column-start: nth($i, 2);
```
Should an offset be bumped up by one place?
e.g. offset of `offset-one` occupy the second column

kingsleyzissou replied:
@piccolazucca brilliant spot. I forgot to update that when moving it to a mixin.

piccolazucca commented on src/_grid.scss:
```diff
@@ -0,0 +1,93 @@
+    }
+}
+
+/* Mobile screens */
```
I've always found desktop first more intuitive, defaulting for desktop and specifying for mobile and tablet after. Thoughts?

kingsleyzissou replied:
@piccolazucca Dim said she agrees - so we'll change it to desktop first.

kingsleyzissou commented:
@dimitraz @piccolazucca I made the changes to limit the gutter helper to the grid, switched to desktop first as discussed. I also added flex helpers for all the columns in the grid or a single column.

@piccolazucca please give feedback on the helper names. I went with `.top`, `.middle`, ` .bottom`, `.left`, `.center` & `.right`. Are those intuitive enough or would you make any changes?

dimitraz commented:
@kingsleyzissou feel free 2 merge
